subsys/bluetooth/controller/hci_driver.c

```diff
@@ -244,38 +274,174 @@ void SIGNALLING_Handler(void)
+		 * static addresses.
+		 */
+		BT_ADDR_SET_STATIC(&addr->a);
+
+		addr->type = BT_ADDR_LE_RANDOM;
+		return 1;
+	}
+	return 0;
+}
+
+static int ble_init(struct device *unused)
+{
+	int err = 0;
+	nrf_lf_clock_cfg_t clock_cfg;
+
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+	clock_cfg.lf_clk_source = NRF_LF_CLOCK_SRC_RC;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL
+	clock_cfg.lf_clk_source = NRF_LF_CLOCK_SRC_XTAL;
+#elif CLOCK_CONTROL_NRF_K32SRC_SYNTH
+	clock_cfg.lf_clk_source = NRF_LF_CLOCK_SRC_SYNTH;
+#else
+#error "Clock source is not defined"
+#endif
+
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_500PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_500_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_250PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_250_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_150PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_150_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_100PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_100_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_75PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_75_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_50PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_50_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_30PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_30_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_20PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_20_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_10PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_10_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_5PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_5_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_2PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_2_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_1PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_1_PPM;
+#else
+#error "Clock accuracy is not defined"
+#endif
+	clock_cfg.rc_ctiv = BLE_CONTROLLER_RECOMMENDED_RC_CTIV;
+	clock_cfg.rc_temp_ctiv = BLE_CONTROLLER_RECOMMENDED_RC_TEMP_CTIV;
+
+	err = ble_controller_init(blectlr_assertion_handler,
+				  &clock_cfg,
+				  SWI5_IRQn);
+	return err;
+}
+
+static int ble_enable(void)
+{
+	int err;
+	ble_controller_cfg_t cfg;
+
+	cfg.master_count.count = 1;
+
+	err = ble_controller_cfg_set(BLE_CONTROLLER_DEFAULT_RESOURCE_CFG_TAG,
+				     BLE_CONTROLLER_CFG_TYPE_MASTER_COUNT,
+				     &cfg);
+	if (err < 0 || err > sizeof(ble_controller_mempool)) {
+		return err;
+	}
+
+	cfg.slave_count.count = 1;
+
+	err = ble_controller_cfg_set(BLE_CONTROLLER_DEFAULT_RESOURCE_CFG_TAG,
+				     BLE_CONTROLLER_CFG_TYPE_SLAVE_COUNT,
+				     &cfg);
+	if (err < 0 || err > sizeof(ble_controller_mempool)) {
+		return err;
+	}
+
+	cfg.buffer_cfg.rx_packet_size = 251;
+	cfg.buffer_cfg.tx_packet_size = 251;
+	cfg.buffer_cfg.rx_packet_count = 10;
+	cfg.buffer_cfg.tx_packet_count = 10;
+
+	err = ble_controller_cfg_set(BLE_CONTROLLER_DEFAULT_RESOURCE_CFG_TAG,
+				     BLE_CONTROLLER_CFG_TYPE_BUFFER_CFG,
+				     &cfg);
+	if (err < 0 || err > sizeof(ble_controller_mempool)) {
+		return err;
+	}
+
+	cfg.event_length.event_length_us = 7500;
```

Comment:
Shouldn't this be configurable?

Reply:
Yes, but as noted earlier, we would like to improve the memory and config in the next round.